hera_qm/xrfi.py

```diff
@@ -1469,79 +1544,166 @@ def xrfi_run(ocalfits_file=None, acalfits_file=None, model_file=None, data_file=
-        uvf_ax, uvf_axf = xrfi_pipe(uvc_a, alg='detrend_medfilt', Kt=kt_size, Kf=kf_size, xants=xants,
-                                    cal_mode='tot_chisq', sig_init=sig_init, sig_adj=sig_adj,
-                                    label='Abscal chisq, round 1.',
-                                    run_check=run_check,
-                                    check_extra=check_extra,
-                                    run_check_acceptability=run_check_acceptability)
-        metrics += [uvf_ag, uvf_ax]
-        flags += [uvf_agf, uvf_axf, uvf_apriori]
-
-        # Calculate metric on omnical data
+    if ocalfits_file is not None:
         uvc_o = UVCal()
-        uvc_o.read_calfits(ocalfits_file)
-        flag_apply(uvf_apriori, uvc_o, keep_existing=True, run_check=run_check,
-                   check_extra=check_extra,
-                   run_check_acceptability=run_check_acceptability)
-        uvf_og, uvf_ogf = xrfi_pipe(uvc_o, alg='detrend_medfilt', Kt=kt_size, Kf=kf_size, xants=xants,
-                                    cal_mode='gain', sig_init=sig_init, sig_adj=sig_adj,
-                                    label='Omnical gains, round 1.',
-                                    run_check=run_check,
-                                    check_extra=check_extra,
-                                    run_check_acceptability=run_check_acceptability)
-        uvf_ox, uvf_oxf = xrfi_pipe(uvc_o, alg='detrend_medfilt', Kt=kt_size, Kf=kf_size, xants=xants,
-                                    cal_mode='tot_chisq', sig_init=sig_init, sig_adj=sig_adj,
-                                    label='Omnical chisq, round 1.',
-                                    run_check=run_check,
-                                    check_extra=check_extra,
-                                    run_check_acceptability=run_check_acceptability)
-        metrics += [uvf_og, uvf_ox]
-        flags += [uvf_ogf, uvf_oxf]
+        uvc_o.read_calfits(acalfits_file)
+        uvf_apriori = UVFlag(uvc_o, mode='flag', copy_flags=True, label='A priori flags.')
         uvf_apriori.to_waterfall(method='and', keep_pol=False, run_check=run_check,
                                  check_extra=check_extra,
                                  run_check_acceptability=run_check_acceptability)
         flags += [uvf_apriori]
-        # Get the absolute chi-squared values
-        uvf_chisq, uvf_chisq_f = chi_sq_pipe(uvc_o, alg='zscore_full_array', modified=True,
-                                             sig_init=sig_init, sig_adj=sig_adj,
-                                             label='Renormalized chisq, round 1.',
-                                             run_check=run_check,
-                                             check_extra=check_extra,
-                                             run_check_acceptability=run_check_acceptability)
-        metrics += [uvf_chisq]
-        flags += [uvf_chisq_f]
+        if omnical_median_filter:
```

Comment:
I like this new organization a lot, but I do wonder whether this would be easier to read and maintain if you make a few helper functions to modularize this whole function better. Right now it's very long and there's a lot of copy-pasted code.

I'm also OK if you make this an issue and defer it to a future PR if that makes more sense to you.

Reply:
I agree that this could probably be more modularized. I think we should defer to a different PR. I've made an issue #346